cpp/src/io/parquet/chunked_state.hpp

```diff
@@ -40,15 +43,24 @@ struct pq_chunked_state {
   /// Overall file metadata.  Filled in during the process and written during write_chunked_end()
   cudf::io::parquet::FileMetaData md;
   /// current write position for rowgroups/chunks
-  size_t current_chunk_offset;
+  std::size_t current_chunk_offset;
   /// optional user metadata
-  table_metadata const* user_metadata = nullptr;
-  /// only used in the write_chunked() case. copied from the (optionally) user supplied
-  /// argument to write_parquet_chunked_begin()
   table_metadata_with_nullability user_metadata_with_nullability;
   /// special parameter only used by detail::write() to indicate that we are guaranteeing
   /// a single table write.  this enables some internal optimizations.
-  bool single_write_mode = false;
+  table_metadata const* user_metadata;
+  /// only used in the write_chunked() case. copied from the (optionally) user supplied
+  /// argument to write_parquet_chunked_begin()
+  bool single_write_mode;
+
+  pq_chunked_state() = default;
+
+  pq_chunked_state(table_metadata const* metadata,
+                   SingleWriteMode mode = SingleWriteMode::NO,
+                   cudaStream_t str     = 0)
+    : user_metadata{metadata}, single_write_mode{mode == SingleWriteMode::YES}, stream{str}
```

Comment:
From Olivier's comment above

Looks like the constructor flips the default value from false to true, which I think is backwards.   Since the default use case for this is indeed "chunked" (multi-write) it should be set that way.   The single-write flag is there so that non-chunked write can quietly reuse the chunked-write code internally.  

Summary : the constructor should set single_write_mode to SingleWriteMode::NO
```suggestion
    : user_metadata{metadata}, single_write_mode{mode == SingleWriteMode::NO}, stream{str}
```

Reply:
@nvdbaranec This initialization list in the ctor only gets called if invoked. And if the user passes `SingleWriteMode::Yes` then ultimately we want the boolean to be set to true. This change would make it set to `false`. If we want the default to be `false` when the default is used, we should be changed the default in the parameter list of the function definition. I will make that change.

Reply:
Ugh, you're right.  I read this backwards.  Sorry :)